community/cypher/compatibility-suite/src/main/java/cypher/feature/parser/matchers/MapMatcher.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package cypher.feature.parser.matchers;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Set;
+
+public class MapMatcher implements ValueMatcher
+{
+    public static final MapMatcher EMPTY = new MapMatcher( Collections.emptyMap() );
+
+    private final Map<String,ValueMatcher> map;
+
+    public MapMatcher( Map<String,ValueMatcher> map )
+    {
+        this.map = map;
+    }
+
+    @Override
+    public boolean matches( Object value )
+    {
+        if ( value instanceof Map )
+        {
+            Map<String,Object> realMap = (Map<String,Object>) value;
+            Set<String> expectedKeys = map.keySet();
+            Set<String> actualKeys = realMap.keySet();
+            boolean match = expectedKeys.equals( actualKeys );
+
+            if ( match )
+            {
+                for ( String key : expectedKeys )
```

Comment:
why not iterate over the entries of `this.map` instead, to avoid one map.get() call per iteration